test/LowerColorado_TX_v4/run_BMI_Coastal.py

```diff
@@ -0,0 +1,276 @@
+    print('Initialize t-route Module...')
+    troute = bmi_troute.bmi_troute()
+    troute.initialize(bmi_cfg_file = base_path + config_dict[config]['yaml'])
+
+    value_dict = {}
+
+    #network = HYFeaturesNetwork(troute._model._supernetwork_parameters,\
+    #                            troute._model._waterbody_parameters,\
+    #                            troute._model._data_assimilation_parameters,\
+    #                            troute._model._restart_parameters,\
+    #                            troute._model._compute_parameters,\
+    #                            troute._model._forcing_parameters,\
+    #                            troute._model._hybrid_parameters,\
+    #                            troute._model._preprocessing_parameters,\
+    #                            troute._model._output_parameters,\
+    #                            verbose=True,\
+    #                            showtiming=troute._model.showtiming,\
+    #                            from_files = False,\
+    #                            value_dict=value_dict) 
+    # 
+    #coastalFile = network.forcing_parameters.get('coastal_boundary_input_file', None)
+
+    coastalFile = 'boundary_forcing/FlowFM_0000_his.nc'
+
+    coastalDataFrame = read_coastal_output(coastalFile)
+
+    # get time reference for coastal BMI transport
+    start_time_coastal = time.time()
+    # capture this as reference time for coastal data for BMI transport of timestamp data
+    timeRef = datetime.fromtimestamp(start_time_coastal)
+    coastal_timeRef =  timeRef.replace(microsecond=0)
+
+    ( timeArray_coastal, nTimes_coastal, stationArray_coastal, stationStringLengthArray_coastal, nStations_coastal) \
+                    = df2a._time_stations_from_df(coastalDataFrame, coastal_timeRef)
+
+    # flatten the actual coastal depth dataframe into a numpy ndarray
+    depthArray_coastal = df2a._flatten_array(coastalDataFrame, np.float64)
+
+    troute.set_value('timeArray_coastal', timeArray_coastal)
+    troute.set_value('nTimes_coastal', nTimes_coastal)
+    troute.set_value('stationArray_coastal', stationArray_coastal)
+    troute.set_value('stationStringLengthArray_coastal', stationStringLengthArray_coastal)
+    troute.set_value('nStations_coastal', nStations_coastal)
+    troute.set_value('coastal_timeRef', coastal_timeRef)
+    troute.set_value('depthArray_coastal', depthArray_coastal)
+
+
+    value_dict['coastal_waterlevelArray'] = np.array([1.0, 2.0, 3.0, 4.0, 5.0], dtype=np.float64)
+    value_dict['coastal_bedlevelArray'] = np.array([12.0, 12.0, 13.0, 14.0, 15.0], dtype=np.float64)
+    value_dict['coastal_nexusID_array'] = np.array([0, 0, 0, 1, 1, 1, 2, 2, 2, 3, 3, 3, 4, 4, 4, 5, 5, 5], dtype=np.int16)
+    value_dict['coastal_nexusID_StrLength_array'] = np.array([3, 3, 3, 3, 3], dtype=np.int16)
+    value_dict['coastal_timeArray'] = np.array([100000, 100000, 100000, 100000, 100000], dtype=np.int16)
+    value_dict['coastal_timeRef'] = 100
```

Comment:
are these being used somewhere? Or was this a placeholder?

Reply:
Removed the value_dict and other debugging statements.

Reply:
Implemented station IDs as numpy int 64 array